source/dem/velocity_verlet_integrator.cc

```diff
@@ -155,92 +292,161 @@ VelocityVerletIntegrator<dim>::integrate(
-                    particle_position[0] +=
-                      particle_properties[PropertiesIndex::v_x] * dt;
-                    particle_position[1] +=
-                      particle_properties[PropertiesIndex::v_y] * dt;
-                    particle_position[2] +=
-                      particle_properties[PropertiesIndex::v_z] * dt;
-
-                    // Updating angular velocity
-                    particle_properties[PropertiesIndex::omega_x] +=
-                      particle_torque[0] * dt_MOI_inverse;
-                    particle_properties[PropertiesIndex::omega_y] +=
-                      particle_torque[1] * dt_MOI_inverse;
-                    particle_properties[PropertiesIndex::omega_z] +=
-                      particle_torque[2] * dt_MOI_inverse;
-                  }
-
-                  // Reinitialize force and torque of particle
-                  particle_force  = 0;
-                  particle_torque = 0;
-
-                  if constexpr (dim == 3)
-                    particle.set_location(particle_position);
-
-                  if constexpr (dim == 2)
+                  // When mobile, the average velocity of the cell and the
+                  // acceleration is updated at the same step that the particles
+                  // are integrated of performance reasons
+                  average_velocities.clear();
+                  average_a_dt.clear();
+
+                  for (auto &particle : particles_in_cell)
                     {
-                      Point<2> position_2d;
-                      position_2d[0] = particle_position[0];
-                      position_2d[1] = particle_position[1];
-                      particle.set_location(position_2d);
+                      types::particle_index particle_id =
+                        particle.get_local_index();
+                      auto particle_properties = particle.get_properties();
+
+                      Tensor<1, 3> &particle_torque = torque[particle_id];
+                      Tensor<1, 3> &particle_force  = force[particle_id];
+
+                      double dt_mass_inverse =
+                        dt / particle_properties[PropertiesIndex::mass];
+                      double dt_MOI_inverse = dt / MOI[particle_id];
+
+                      particle_position = [&] {
+                        if constexpr (dim == 3)
+                          {
+                            return particle.get_location();
+                          }
+                        else
+                          {
+                            return (point_nd_to_3d(particle.get_location()));
+                          }
+                      }();
+
+                      // Calculate acceleration * dt and add velocity value for
+                      // average acceleration computation
+                      Tensor<1, 3> a_dt =
+                        dt_g + particle_force * dt_mass_inverse;
+                      average_a_dt += a_dt;
```

Comment:
I prefer to use "acc" as a variable name in these situations and then calculate the average, for safety. But it is up to you.